javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/CompilationUnitContext.java

```diff
@@ -188,7 +208,7 @@ public CompilationUnitContext(CompilationUnit wrappedNode, TypeSolver typeSolver
 
         // Look in the java.lang package
         SymbolReference<ResolvedReferenceTypeDeclaration> ref = typeSolver.tryToSolveType("java.lang." + name);
-        if (ref.isSolved()) {
+        if (ref != null && ref.isSolved()) {
```

Comment:
why should that be null? I think it should not

Reply:
Ok, my take is this:

The method `TypeSolver#tryToSolveType(String)` makes no guarantees that it never returns `null`. It is not annotated with an annotation such as `@NotNull`, nor is there a Javadoc comment explaining that this method never returns `null`. I didn't write this method, so for me there is no way to know if, perhaps, this method indeed never returns `null`, but this guarantee is simply undocumented. By default, lacking any documentation, I have to assume that it *could* return `null`.

Generally speaking, I think that if a method makes no guarantees that it does not return `null`, then we cannot simply assume that it will not return `null`. It's better to be safe than sorry.

I just tried to remove the `ref != null` checks that I added, but then there is a failing test: `ContextTest.resolveReferenceUsingQualifiedName()`. So indeed, it seems like the method  `TypeSolver#tryToSolveType(String)` sometimes returns `null`. Also, if you look around in the same method, there are `ref != null && ref.isSolved()` checks in various places that you wrote yourself. ;-) So it appears to be a valid paradigm. I simply added the `ref != null` check in those places where it was missing.

Reply:
I understand your position, let's merge this.

However no method in Javaparser is supposed to return null, so from my point of view if something return null is a bug. I will open a separate ticket to verify `TypeSolver#tryToSolveType(String)` does not return null and correct where it does so. Thank you for pointing out this